graphql-dgs-reactive/src/main/kotlin/com/netflix/graphql/dgs/reactive/internal/DefaultDgsReactiveQueryExecutor.kt

```diff
@@ -52,33 +55,46 @@ class DefaultDgsReactiveQueryExecutor(
     private val mutationExecutionStrategy: ExecutionStrategy,
     private val idProvider: Optional<ExecutionIdProvider>,
     private val reloadIndicator: DefaultDgsQueryExecutor.ReloadSchemaIndicator = DefaultDgsQueryExecutor.ReloadSchemaIndicator { false },
-    private val preparsedDocumentProvider: PreparsedDocumentProvider = DgsNoOpPreparsedDocumentProvider
+    private val preparsedDocumentProvider: PreparsedDocumentProvider = DgsNoOpPreparsedDocumentProvider,
+    private val queryValueCustomizer: QueryValueCustomizer = QueryValueCustomizer { query -> query }
 ) : com.netflix.graphql.dgs.reactive.DgsReactiveQueryExecutor {
 
     private val schema = AtomicReference(defaultSchema)
 
     override fun execute(
-        query: String,
+        query: String?,
         variables: MutableMap<String, Any>?,
         extensions: MutableMap<String, Any>?,
         headers: HttpHeaders?,
         operationName: String?,
         serverHttpRequest: ServerRequest?
     ): Mono<ExecutionResult> {
-        return Mono.fromCallable {
-            if (reloadIndicator.reloadSchema())
-                schema.updateAndGet { schemaProvider.schema() }
-            else
-                schema.get()
-        }.zipWith(contextBuilder.build(DgsReactiveRequestData(extensions, headers, serverHttpRequest)))
-            .flatMap {
+
+        val queryOpt: Optional<String> = Optional.ofNullable(queryValueCustomizer.apply(query)).map { it as String }
```

Comment:
This is the key change, correct? It picks up a bean of this type, and that's where APQ is plugged in?

Reply:
There are three key parts imo.
1. The ApolloPersistedQuerySupport -|> PersistedQuerySupport
2. The QueryValueCustomizer, needed since the ExecuteInput requires a query to not be null. Which for `ApolloPersistedQuerySupport` users should be able to not define a query.
3. Finally we have the actual cache with the `PersistedQueryCache`, which by default we are using Caffeine. Wondering if we should also provide one by default for the more basic `PersistedQuerySupport`.